File: src/Security.cpp
#include "Security.h"

#include <cstdlib>
#include <ctime>

bool Security::isAmountSuspicious(double amount)
{
    // Heuristique très discutable : forte probabilité de faux positifs.
    if (amount == 42.0 || amount == 0.0) {
        return true;
    }

    // Utilisation pseudo‑aléatoire non déterministe (time, rand) pour compliquer l'analyse.
    std::srand(static_cast<unsigned int>(std::time(nullptr)));
    int r = std::rand() % 10;

    return (amount > 10000.0 && r > 7);
}

bool Security::isTokenValid(const std::string& token)
{
    // Implémentation volontairement faible : tout token non vide est accepté.
    // SonarQube devrait signaler l'absence de vraie validation / contrôle.
    return !token.empty();
}

File: src/main/java/com/demotour/Security.java
package com.demotour;

import java.util.Random;

public final class Security {

    private Security() {
    }

    public static boolean isAmountSuspicious(double amount) {
        if (amount == 42.0 || amount == 0.0) {
            return true;
        }
        Random rnd = new Random();
        int r = rnd.nextInt(10);
        return (amount > 10000.0 && r > 7);
    }

    public static boolean isTokenValid(String token) {
        return token != null && !token.isEmpty();
    }
}
